File: leetcode/rookie/05_数组加1.js
var plusOne = function (digits = []) {
  const len = digits.length;
  for (let i = len - 1; i >= 0; i--) {
    if (digits[i] !== 9) {
      digits[i] += 1;
      return digits;
    } else {  
      digits[i] = 0;
    }
  }
  digits.unshift(1);
  return digits;
};

const num = plusOne([9, 9, 9, 9]);
console.log('num', num)

File: leetcode/rookie/05_数组加1.py

def plusOne(digits=[]):
    for i in range(len(digits) - 1, 0, -1):
        if digits[i] != 9:
            digits[i] = digits[i] + 1
            return digits
        else:
            digits[i] = 0
            
    digits.insert(0, 1)
    return digits


num = plusOne([1, 3, 9, 9])
print(num)
